fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/DimensionalFilter.java

```diff
@@ -14,26 +18,52 @@
  *
  * @param <T> a DimensionalFilter
  */
+@JsonInclude(JsonInclude.Include.NON_NULL)
 public abstract class DimensionalFilter<T extends DimensionalFilter<? super T>> extends Filter {
 
     private final Dimension dimension;
+    private final ExtractionFunction extractionFunction;
 
     /**
-     * Constructor.
+     * Constructor, default extraction function to the one on dimension if it has one.
      *
      * @param dimension  Dimension to filter
      * @param type Type of the filter
      */
     protected DimensionalFilter(Dimension dimension, FilterType type) {
         super(type);
+
+        this.dimension = dimension;
+        if (dimension instanceof ExtractionFunctionDimension) {
+            extractionFunction = ((ExtractionFunctionDimension) dimension).getExtractionFunction().get();
```

Comment:
Are we sure `.get()` returns a extracFn?

Reply:
or null